### ISSUE TITLE
Serialization of 'Closure' is not allowed

### ISSUE THREAD

**Issue by sroutier** [santigarcor/laratrust#201]
- Laravel Version: 5.5.9
- Laratrust Version: 4.0.3

I have this strange error "Serialization of 'Closure' is not allowed" that occurs as a result of calling ```$user->hasRole('core.admins')```. 
I have tracked it down to the ```Illuminate\Cache\FileStore:put($key, $value, $minutes)``` method which blows up on line 64/65:
```
        $this->files->put(
            $path, $this->expiration($minutes).serialize($value), true
        );

```

The strange things are that:
1) It works in thinker.
2) The variable $value, being serialized is a collection of roles, or at least should be since it was retrieved on by ```Laratrust\Traits\LaratrustUserTrait:cachedRoles()``` when it called ```Cache::remember(...)``` with a closure.

Two questions:
1) As anybody seen this before or would have some highly appreciated pointers?
2) Would there be any appetite for making the caching of permissions and roles optional?

Thank you.
/S

**Comment by santigarcor** [comment]
1. What cache driver are you using?
2. I've thought in making the caching optional, and it's doable. Maybe i can think of implementing it for the 5.0 release.

**Comment by sroutier** [comment]
Using the default file store at the moment, will try others to see as well.
Also I will try using using plain laravel and laratrust to try and get a scenario that is easier to reproduce without all the other dependencies that I have at the moment.
How close to release 5.0 are we?

/s

**Comment by santigarcor** [comment]
I think we are very close. I'm now working on the docs.

**Comment by santigarcor** [comment]
@sroutier Have you found a way to replicate this issue?

**Comment by sroutier** [comment]
Sorry man. I am stuck in a gym all day. Will try tonight or tomorrow.
/s

**Comment by sroutier** [comment]
Thanks for the option to enable or disable the cache. 

Otherwise bad news, for me at least. I tested both the default file store and redis and both fail with the same error above.

I then created a brand new project with the latest packages of Laravel 5.5.13, Laratrust 4.0.5 and all the dependencies that were automatically installed. The call to ```$user->hasRole()``` functioned properly. 
I then used my original composer.json/lock, using Laravel 5.5.9, Laratrust 4.0.3, in this new project and re-installed all packages and ```hasRole()``` still worked properly...

But if I use the "dev-master" branch in my actual project it all just work without any issues, regardless of whether I use caching or not!?

I am going to try to track this down a little more but looks like the dev-master branch is good for me.

Thanks.
/S
